### DISCUSSION TITLE
LDM Notes for 10/10/2022

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#6526] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-10-10.md

## Agenda

- Working set triage
    - Static classes implementing interfaces
    - Null-conditional assignment
    - Lambda default parameters
- Working groups
- `params` support for lambda default parameters
    - Conclusion

The notes for October 5th are still being edited, and will hopefully be up in the next few days. I haven't forgotten about them.

**Comment by theunrepentantgeek** [comment]
> At current rate of working group increase (1 in 2021, 8 in 2022) we will have 64 working groups come October 2023.

Pleased to see all the important numbers are powers of 2.

**Reply by YairHalberstadt** [reply]
I think this estimate is incorrect - working groups should increase linearly, not exponentially, unless working groups start spinning off child working groups...

**Reply by MgSam** [reply]
I'm disappointed that we're left wondering what the median and mode for working group size are.

**Reply by 333fred** [reply]
I was too lazy to calculate those, sorry.

**Reply by orthoxerox** [reply]
You have to do math to find the mean, the median and mode are trivial to find in comparison.

**Reply by 333fred** [reply]
> You have to do math to find the mean, the median and mode are trivial to find in comparison.

Mean is a dead-simple calculation that doesn't actually require me to analyze the data, just read the numbers in order and punch them into a calculator. I have to actually count things for median/mode :P.

**Comment by ghord** [comment]
Is the previously mentioned (LDM 2022-08-24) work on expression trees assigned to any of the working groups?

**Reply by MadsTorgersen** [reply]
Not currently. We have a somewhat clear understanding of the approach we would take, but don't have a good write-up of it. Frankly we are disheartened by the sheer amount of compiler and library work it would entail. That said, I could see one of us spending some time on it to at least get to a rough proposal doc written up with what we've discussed.

**Reply by iam3yal** [reply]
@MadsTorgersen
> I could see one of us spending some time on it to at least get to a rough proposal doc written up with what we've discussed.

That would be great.

**Reply by ghord** [reply]
@MadsTorgersen Thank you for taking time to respond. I feel that Expression trees are such unique feature of C# that it would be shame to abandon them completely. Looking forward to reading the proposal.
